command.go

```diff
@@ -103,11 +103,14 @@ func Auth(secret string) (*Command, error) {
 	return &Command{[]byte("AUTH"), nil, []byte(secret)}, nil
 }
 
-// Register creates a new Command to add a topic/channel for the connected nsqd
-func Register(topic string, channel string) *Command {
-	params := [][]byte{[]byte(topic)}
-	if len(channel) > 0 {
-		params = append(params, []byte(channel))
+// Register creates a new Command to add a topic/channel with state for the connected nsqd
+func Register(topic string, channel string, states ...[]byte) *Command {
```

Comment:
I'm still not particularly happy with this API.

For one, it's a breaking change to `go-nsq` (yes, I realize that it's incredibly unlikely that users are depending on this function to speak to `nsqlookupd` over the TCP protocol, nonetheless).

We could avoid introducing this breaking change by not touching `Register` at all, instead just _adding_ the new `SyncState` command and then sending a `SyncState` command immediately following a new registration (in `nsqd`).

Reply:
Okay, I removed what I added before